libraries/backbone.es6.js

```diff
@@ -28,10 +45,18 @@ define('backbone.es6', [
     // The constructor function for the new subclass is either defined by you
     // (the "constructor" property in your `extend` definition), or defaulted
     // by us to simply call the parent constructor.
-    if (protoProps && _.has(protoProps, 'constructor')) {
-      child = protoProps.constructor;
+    if (hasNativeClassSupport) {
+      if (protoProps && _.has(protoProps, 'constructor')) {
+        child = eval('class e extends protoProps.constructor { }; e;');
+      } else {
+        child = eval('class e extends parent { }; e;');
+      }
     } else {
-      child = function(){ return parent.apply(this, arguments); };
+      if (protoProps && _.has(protoProps, 'constructor')) {
+        child = protoProps.constructor;
+      } else {
+        child = function(){ return parent.apply(this, arguments); };
+      }
```

Comment:
could put all this out into a separate function called `getChild`, then apply the return early pattern to the code and return the child. You could also set this as a var `protoProps && _.has(protoProps, 'constructor')`  to avoid duplication